build.rs

```diff
@@ -39,14 +46,21 @@ fn main() {
         );
 
         println!("cargo:rerun-if-changed=build.rs");
-        println!("cargo:rerun-if-changed=.git/HEAD");
-
-        let head = std::fs::read_to_string(".git/HEAD").unwrap_or_default();
-        if head.starts_with("ref: ") {
-            let path_str = format!(".git/{}", head[5..].trim());
-            let path = Path::new(&path_str);
-            if path.exists() {
-                println!("cargo:rerun-if-changed={}", path_str);
+
+        {
+            let git_head = ".git/HEAD";
```

Comment:
I think we should use `git rev-parse --git-dir` to get the git directory.

Reply:
Allow `git rev-parse --git-dir` failing.

Reply:
Resolved and added a step to the manual test steps in the PR description.